lib/spec/openapi/utils.js

```diff
@@ -323,20 +323,25 @@ function resolveResponse (fastifyResponseJson, produces, ref) {
 
     // add schema when type is not 'null'
     if (rawJsonSchema.type !== 'null') {
-      const content = {}
+      if (resolved.content && !resolved.content.type) {
+        response.content = resolved.content
+      } else {
+        const content = {}
 
-      if ((Array.isArray(produces) && produces.length === 0) || typeof produces === 'undefined') {
-        produces = ['application/json']
-      }
+        if ((Array.isArray(produces) && produces.length === 0) || typeof produces === 'undefined') {
+          produces = ['application/json']
+        }
 
-      delete resolved[xResponseDescription]
+        delete resolved[xResponseDescription]
 
-      const media = schemaToMedia(resolved)
-      produces.forEach((produce) => {
-        content[produce] = media
-      })
+        const media = schemaToMedia(resolved)
 
-      response.content = content
+        for (const produce of produces) {
+          content[produce] = media
+        }
```

Comment:
Why change from `forEach`?

Reply:
for loop seems to be faster than forEach, @Eomm also mentioned that. https://github.com/fastify/fastify-swagger/pull/665#discussion_r968043372